### DISCUSSION TITLE
how to use custom fields from ACF on the page?

### DISCUSSION THREAD

**Discussion by AeShevch** [WebDevStudios/nextjs-wordpress-starter#935] [Q&A]
Greetings!
I can't figure out how to use custom fields from ACF on the page?
I created a custom field with the text type, checked "Show in GraphQL", filled in the field on the page, but nothing similar appeared in the page props.

![Снимок экрана 2022-02-18 в 1 05 01](https://user-images.githubusercontent.com/44289476/154578821-82c955e1-18e7-4c63-ae4c-5d2d6d764580.png)

**Comment by AeShevch** [comment]
@itsamoreh

Sorry for contacting you directly, but I can't find a solution :((

**Reply by itsamoreh** [reply]
@AeShevch Sorry for the late reply! Have you added the ACF fields to your query?

For example, if I have a field group that's set to show on pages:
<img width="1137" alt="screen-capture 2022-04-29 at 15 54 27 " src="https://user-images.githubusercontent.com/12365909/166061320-fbbd6591-e340-48e0-b601-9767d3a59359.png">

Your GraphQL query for page routes should look something like this:
```
query PageByID {
  pageBy(pageId: 383) {
    id
    slug
    theCustomFieldName {
      aField
    }
  }
}
```

The result should look something like this:
```
{
  "data": {
    "pageBy": {
      "id": "cG9zdDozODM=",
      "slug": "home",
      "theCustomFieldName": {
        "aField": "This is the custom field!"
      }
    }
  },
 ...
}
```
